sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkStringTypeUtils.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.util
+
+import org.apache.spark.sql.internal.SqlApiConf
+
+trait SparkStringTypeUtils {
+  /**
+   * Check if default collation is up-to-date with a config.
+   */
+  private def refreshDefaultCollation(): Unit = {
+    if (SqlApiConf.get.defaultCollation != DEFAULT_COLLATION) {
+      DEFAULT_COLLATION = SqlApiConf.get.defaultCollation
+      DEFAULT_COLLATION_ID = CollationFactory.collationNameToId(DEFAULT_COLLATION)
```

Comment:
This is just a hash map lookup, is it really worth the complexity to build a cache for the default collation id?

Reply:
I would just like to make sure that we never end up calling this on execution side. E.g. if we end up making hash map lookup per row we are dead. Could we use `AnalysisHelper.inAnalyzer` flag to make sure that this field is only accessed during analysis?

This is a healthy thing to do for Delta side as well - Delta should never rely on session level collation (e.g. for constraints/computed columns) so it should be good if we can actually enforce this.

Reply:
to enforce it's only called at the driver side, we can assert `SparkEnv.get.executorId == SparkContext.DRIVER_IDENTIFIER`

Reply:
Could you take a look at how I added this check, as a lot of tests are failing for reason SparkEnv.get == null